ansible_base/lib/abstract_models/common.py

```diff
@@ -127,7 +99,7 @@ def save(self, *args, non_existent_user_fatal=True, **kwargs):
 
         # Manually perform auto_now_add and auto_now logic.
         now = timezone.now()
-        user = self._attributable_user(non_existent_user_fatal)
+        user = get_current_user() or get_system_user()
```

Comment:
This was not my intention.  I think model saves that are done via non-web-request processes should be distinguished from non-authenticated web requests.  This does not do that:

| `get_current_user()` | `user` | intention |
|----|----|----|
| some `User()` | `User()` | that `User()` |
| `AnonymousUser()` | `AnonymousUser()` | `User(system)`, since it is (probably) auth |
| `None` | `User(system)` | `None`, since this was non-web |

Reply:
Unless I'm wrong about `AnonymousUser` evaluating as truthy.  Still, I wouldn't count on it.

Reply:
This would convert None to system_user but AnonymousUser would fall through and eventually cause an exception. The way I was taking this solution, it's up to the models to grab AnonymousUser and choose if its appropriate to sub out system in place of Anonymous. Let me know if you are good with this approach. I personally had concerns with just converting Anonymous into System incase there was some kind of upstream coding issue that allowed an Anonymous request to attempt to save something.

Reply:
I very strongly do not believe that high-level validation of this sort should be taking place in the `.save()` method.  Yes, Django does have the concept of validation that can take place there, but those are all pretty low level simple things.

Reply:
`.save()` is not the place to determine if a developer left a view unguarded.